engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -164,8 +164,18 @@ object WorkflowExecutionActor {
     }
 
     private def arePrerequisitesDone(key: JobKey): Boolean = {
-      val upstream = key.scope.prerequisiteScopes.toList.map(s => upstreamEntries(key, s))
-      val downstream = key match {
+      val upstream = key.scope match {
+        case node: GraphNode => node.upstream collect {
+          // Only scatters and calls are in the execution store for now (not declarations)
+          // However declarations are nodes so they can be an upstream dependency
+          // We don't want to look for those in the execution store (yet ?) since upstreamEntry would return None
```

Comment:
yes the wdl4s `upstream` gives you that, I was talking about the `upstreamEntry` in the WEA that tries to find an upstream dependency in the execution store. However we are not adding Declarations to the executions store (yet - I think we should but this PR is focused on feature party, there is a ticket for adding declarations).